### PR TITLE
Added support for Balance (Astral Power)

### PR DESCRIPTION
**Pull request by EricaPomme**
Hello!

Just sharing my quick adaptation to add support for Balance Druid.

It plays tones 1, 3, and 4 when your next cast will put you at 30, 60, or 90 Astral Power (to allow for planning Starsurges), and tone 5 when you're at or over cap. Tone 2 is set at 50 for Starfall, but currently commented out.

Sorry if this is messy, I'm not super familiar with Lua, so I just drew on what I could see from other functions in the add-on.

### CODE REVIEW

**Comment by TravisSpomer** [comment]
Hi! Thanks for the contribution. I haven't missed it—I'll take a look soon. Just been really busy. :)

File: HearKitty.lua [hunk 1]
```diff
@@ -275,6 +277,57 @@ function KittyOnInsanityChange()
 	end
 end
 
+function KittyOnAstralPowerChange()
+	local AstralPower = UnitPower("player", Enum.PowerType.AstralPower)
+
+	-- Get relevant talent, current Eclipse states
+	local _, _, _, SoulOfTheForestSelected = GetTalentInfo(5, 1, 1)
+	local LunarActive, SolarActive = false, false
+	for i = 1, 40 do
```
Comment (TravisSpomer) [inline HearKitty.lua:286]:
```suggestion
	for i = 1, BUFF_MAX_DISPLAY do
		local i
```

**Review by TravisSpomer — APPROVED** [review]
This looks great! Thanks for working hard to maintain the existing code style. I'll make a couple of minor tweaks, approve this, and release 1.9.3.

**Comment by TravisSpomer** [comment]
By the way, I'm guessing that this plays tones 2 and 3 whenever you get to 60 AP—if you just want one tone there, we should be able to easily do that by changing `KittyCurrentMaxStacks`. I'll let you be the judge of what makes most sense; just let me know and I can switch it up!

**Comment by EricaPomme** [comment]
> By the way, I'm guessing that this plays tones 2 and 3 whenever you get to 60 AP—if you just want one tone there, we should be able to easily do that by changing `KittyCurrentMaxStacks`. I'll let you be the judge of what makes most sense; just let me know and I can switch it up!

That sounds like preferable behaviour, and not something I had caught. Good idea!

Also, thank you for making your code so well commented and easy to read. It makes it a lot easier for newbies like me to stick our noses into projects.

**Comment by TravisSpomer** [comment]
I do definitely opt for readability over "proper" architecture, especially since pretty much all of my addons were hacked together more than 10 years ago when I knew nothing about Lua. 🙂 Usually that readability is just *for me a year or two from now* anyway when I won't remember how any of the code works, since I'm pretty forgetful.

Mind testing it out so I don't have to spec out a balance druid and figure out how to play it? Just reduce 3, 4, and 5 by 1...

```lua
	if AstralPower >= 100 - APGainOffset then
		AstralPower = 4
	elseif AstralPower >= 90 - APGainOffset then
		AstralPower = 3
	elseif AstralPower >= 60 - APGainOffset then
		AstralPower = 2
```

...and drop `KittyCurrentMaxStacks` by 1 as well...

```lua
	if (AstralPower ~= KittyLastSoundPlayed) then
		-- (No-op if the number actually hasn't changed.)
		KittyCurrentMaxStacks = 4
```

Someday I'll make a special unique set of tones for resources that cap at 4 and 6 instead of just reusing the 5-point ones, and this will work with that too.

Let me know how it feels, and if you like it, just let me know and I'll change it. No need to even open a PR!

**Pull request merged**